### ISSUE TITLE
[Accessibility] Narrator cannot announce the indeterminate state of CheckBox correctly 

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#9157]
### .NET version

.Net 8.0 SDK build: 8.0.100-preview.5.23268.13

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, it is a regression issue, cannot repro in .Net Core 3.1, and repro in .Net 5, 6, 7 versions

### Issue description

When set CheckBox's CheckState to **Indeterminate**, Narrator cannot announce that state is **Indeterminate**, but "Checked".

**.Net 8.0:**
![image](https://github.com/dotnet/winforms/assets/26474449/2352ea55-16d7-48c5-bad1-8b25df310ee9)
![image](https://github.com/dotnet/winforms/assets/26474449/45ed5dbb-95c6-400f-9a1f-90f83346f1c2)

**.Net Core 3.1:**
![image](https://github.com/dotnet/winforms/assets/26474449/b9246b17-0dba-4c65-9e3c-1ca5a656624b)
![image](https://github.com/dotnet/winforms/assets/26474449/08deb3a7-12c5-46e6-807a-370c30ac02cf)

**.Net framework 4.8.1:**
![image](https://github.com/dotnet/winforms/assets/26474449/df62153a-eb1a-4d47-a92b-e700c83802d7)
![image](https://github.com/dotnet/winforms/assets/26474449/62d8b750-1ee8-4c6c-88cd-09b248585146)

### Steps to reproduce

**Repro steps:**
1. Create a Winforms .Net Core application with CheckBox
2. Set CheckBox's CheckState to **Indeterminate**
3. Build and run application
4. Launch Narrator to focus on CheckBox

or use this sample application:
[WinFormsApp19.zip](https://github.com/dotnet/winforms/files/11528028/WinFormsApp19.zip)

**Comment by Tanya-Solyanik** [comment]
@dmitrii-drobotov if the fix is available sooner, we can service it into 8.0 preview5

**Comment by dmitrii-drobotov** [comment]
* We should also look into other controls with Toggle pattern and make a fix for them as well, if they have the same problem. I can see that at least CheckedListBox has the same issue, when adding items this way: `checkedListBox1.Items.Add("item", CheckState.Indeterminate);`.
* It would be good to also update our test apps to include indeterminate state for controls that support it.

**Comment by v-elnovikova** [comment]
> * We should also look into other controls with Toggle pattern

Found similar issues:
- [#9243](https://github.com/dotnet/winforms/issues/9243) for CheckedListBox
- [#9256](https://github.com/dotnet/winforms/issues/9256) for DataGridViewCheckBoxCell

**Comment by John-Qiao** [comment]
Verified with .NET SDK 8.0.100-preview.6.23315.26 build from main branch, this issue is fixed that Narrator can announces the **Indeterminate** state now.

![TestResult1](https://github.com/dotnet/winforms/assets/45864985/35a17ba7-bcca-4abd-a1fe-782de41a9b3d)
![TestResult2](https://github.com/dotnet/winforms/assets/45864985/9ca4c9a6-ede4-478a-aa24-e7c1a944cd0c)

**Comment by Amy-Li03** [comment]
Verified with .NET SDK 8.0.100-preview.6.23327.6 test pass build, this issue is fixed, test result is same as above screenshot.
